MediaBrowser.Providers/MediaInfo/AudioResolver.cs

```diff
@@ -0,0 +1,164 @@
+#pragma warning disable CA1002, CS1591
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Emby.Naming.Audio;
+using Emby.Naming.Common;
+using Jellyfin.Extensions;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.MediaEncoding;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Dlna;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.MediaInfo;
+
+namespace MediaBrowser.Providers.MediaInfo
+{
+    public class AudioResolver
+    {
+        public async Task<List<MediaStream>> GetExternalAudioStreams(
+            Video video,
+            int startIndex,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache,
+            ILocalizationManager localizationManager,
+            IMediaEncoder mediaEncoder,
+            CancellationToken cancellationToken)
+        {
+            var streams = new List<MediaStream>();
+
+            if (!video.IsFileProtocol)
+            {
+                return streams;
+            }
+
+            List<string> paths = GetExternalAudioFiles(video, directoryService, namingOptions, clearCache);
+
+            await AddExternalAudioStreams(streams, paths, startIndex, localizationManager, mediaEncoder, cancellationToken);
+
+            return streams;
+        }
+
+        public List<string> GetExternalAudioFiles(
+            Video video,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache)
+        {
+            List<string> paths = new List<string>();
+
+            if (!video.IsFileProtocol)
+            {
+                return paths;
+            }
+
+            paths.AddRange(GetAudioFilesFromFolder(video.ContainingFolderPath, video.Path, directoryService, namingOptions, clearCache));
+            paths.AddRange(GetAudioFilesFromFolder(video.GetInternalMetadataPath(), video.Path, directoryService, namingOptions, clearCache));
+
+            return paths;
+        }
+
+        private List<string> GetAudioFilesFromFolder(
+            string folder,
+            string videoFileName,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache)
+        {
+            List<string> paths = new List<string>();
+            string videoFileNameWithoutExtension = Path.GetFileNameWithoutExtension(videoFileName);
+
+            if (!Directory.Exists(folder))
+            {
+                return paths;
+            }
+
```

Comment:
No need to allocate if we're gonna exit early
```suggestion
            if (!Directory.Exists(folder))
            {
                return paths;
            }

            List<string> paths = new List<string>();
            string videoFileNameWithoutExtension = Path.GetFileNameWithoutExtension(videoFileName);
```

Reply:
I'll implement this manually since the List<string> paths = ... must be done before the if(!Directory.Exists(folder)) :)

Reply:
Right, sorry. With the return type changed to `IEnumerable` you can use `Enumerable.Empty<string>()`.